File: servicos/faturamento/ServicoFaturamento.Api/Dominio/NotaFiscal.cs
namespace ServicoFaturamento.Api.Dominio;

public sealed class NotaFiscal
{
    public long Numero { get; set; }

    public StatusNotaFiscal Status { get; set; } = StatusNotaFiscal.Aberta;

    public DateTime CriadaEmUtc { get; set; } = DateTime.UtcNow;

    public List<ItemNotaFiscal> Itens { get; set; } = [];
}

File: interface/src/app/modelos/nota-fiscal.ts
export interface ItemNotaFiscal {
  produtoId: number;
  quantidade: number;
}

export interface NotaFiscal {
  numero: number;
  status: string;
  criadaEmUtc: string;
  itens: ItemNotaFiscal[];
}

export interface CriarNotaFiscalRequisicao {
  itens: ItemNotaFiscal[];
}
